src/core/scene/a-scene.js

```diff
@@ -128,20 +133,15 @@ var AScene = module.exports = registerElement('a-scene', {
      */
     enterVR: {
       value: function (event) {
-        this.setStereoRenderer();
-        if (isMobile) {
-          setFullscreen(this.canvas);
-        } else {
-          this.stereoRenderer.setFullScreen(true);
-        }
+        if (!vrDisplay) { return; }
+        vrDisplay.requestPresent([{source: this.renderer.domElement}]);
         this.addState('vr-mode');
```

Comment:
we should not blindly assume that this succeeded. we should change state and emit events only when the promise resolves à la https://github.com/cvan/aframe/commit/af2cfa39fe8bc72cc1875009f810f7e3c8560f13#diff-eb9bb1084863a2f65678ea67d6e7cf5eR135